cocotb/scheduler.py

```diff
@@ -210,10 +210,9 @@ class Scheduler(object):
     _MODE_WRITE    = 3  # noqa
     _MODE_TERM     = 4  # noqa
 
-    # Singleton events, recycled to avoid spurious object creation
-    _next_time_step = NextTimeStep()
-    _read_write = ReadWrite()
-    _read_only = ReadOnly()
```

Comment:
I'd be inclined to keep these. Within the scheduler itself, every bit of optimization counts much more than it does in user or driver code.

Reply:
Let's see what profiling says the effect is before reverting this commit - thanks for keeping it separate